src/common/plugin/CurrencyPlugin.ts

```diff
@@ -81,6 +86,16 @@ export function makeCurrencyPlugin(
         throw e
       })
       return currencyTools
+    },
+
+    async updateInfoPayload(infoPayload) {
+      try {
+        cachedInfoPayload = asInfoPayload(infoPayload)
+      } catch (_) {
+        log.warn('invalid infoPayload', infoPayload)
+      }
```

Comment:
The plugin state object has a member called `getInfoPayload`, but this seems weirdly indirect. The plugin state object could just have an `infoPayload` member, which is the value itself.

Then here we can do `pluginState.infoPayload = asInfoPayload(infoPayload)` to update the value directly, and in the constructor we can just pass `infoPayload` as a value. Fixing this would remove a level of indirection.

However, I understand that this is just a style thing, so if you disagree, feel free to ignore. The way you have it certainly works.